File: viasoft.licensing.customerlicensing-frontend/src/client/customer-licensing/model/ProductOutput.ts
import { ProductStatus } from "src/app/common/enums/ProductStatus";
import { ProductType } from "src/app/common/enums/ProductType";
import { LicensingModels } from "./LicensingModels";
import { LicensingModes } from "./LicensingModes";

export interface ProductOutput {
    id?: string | null;
    name?: string | null;
    identifier?: string | null;
    isActive?: boolean | null;
    softwareId?: string | null;
    softwareName?: string | null;
    numberOfLicenses?: number | null;
    licensingModel?: LicensingModels | null;
    licensingMode?: LicensingModes | null;
    numberOfUsedLicenses?: number | null;
    productType?: ProductType | null;
    status?: ProductStatus | null;
}

File: viasoft.licensing.customerlicensing/Viasoft.Licensing.CustomerLicensing.Domain/DTOs/Product/ProductOutput.cs
using System;
using Viasoft.Licensing.CustomerLicensing.Domain.Enums;

namespace Viasoft.Licensing.CustomerLicensing.Domain.DTOs.Product
{
    public class ProductOutput
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Identifier { get; set; }
        public bool IsActive { get; set; }
        public Guid SoftwareId { get; set; }
        public string SoftwareName { get; set; }
        public int NumberOfLicenses { get; set; }
        public LicensingModels LicensingModel { get; set; }
        public LicensingModes? LicensingMode { get; set; }
        public ProductStatus Status { get; set; }
        public int NumberOfUsedLicenses { get; set; }
        public ProductType ProductType { get; set; }
    }
}